File: LeetCode/Simulation/Python/Leetcode769-max-chunks-to-make-sorted.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        n, res, mx = len(arr), 0, 0
        for k,v in enumerate(arr):
            mx = max(mx, v)
            res += 1 if mx == k else 0
        return res

File: LeetCode/Simulation/Rust/Leetcode769-max-chunks-to-make-sorted.rs
impl Solution {
    pub fn max_chunks_to_sorted(arr: Vec<i32>) -> i32 {
        let n = arr.len();
        let (mut res, mut mx) = (0,0);
        for i in 0..n {
            mx = mx.max(arr[i]);
            if mx == i as i32 {
                res += 1;
            }
        }
        res
    }
}
